### PR TITLE
Contact And Lead Mapping.

### PR DESCRIPTION
**Pull request by MehreenT**
1) Updated the `OAuth Controller` and RSpec test cases for fixing the `key` issue.
2) Updated `Person/ContactAndLead` mapping with `sub entities`.
3) Updated test cases for `Person/ContactAndLead` mapping with `sub entities`.
4) Updated `Person/Lead` mapping & RSpec test cases after `End-to-End Testing`.

### CODE REVIEW

File: app/controllers/oauth_controller.rb [hunk 1]
```diff
@@ -20,8 +20,8 @@ def create_omniauth
       auth = request.env["omniauth.auth"]
       current_organization.oauth_provider = auth.provider
       current_organization.oauth_uid = auth.uid
-      current_organization.oauth_token = auth.credentials.token
-      current_organization.refresh_token = auth.credentials.refresh_token
+      current_organization.encrypted_oauth_token = auth.credentials.token
```
Comment (MarcoBgn) [inline app/controllers/oauth_controller.rb]: You should set these values to the non-encrypted fields, they will be encrypted by the gem

File: app/models/maestrano/connector/rails/external.rb [hunk 2]
```diff
@@ -17,8 +17,8 @@ def self.create_account_link(organization = nil)
   end
 
   def self.get_client(organization)
-    Restforce.new :oauth_token => organization.oauth_token,
-      refresh_token: organization.refresh_token,
+    Restforce.new :oauth_token => organization.encrypted_oauth_token,
```
Comment (MarcoBgn) [inline app/models/maestrano/connector/rails/external.rb]: Same here for the encrypted values

File: app/controllers/oauth_controller.rb (hunk 1 above)
Comment (MehreenT) [reply app/controllers/oauth_controller.rb]: Thankyou. Feedback incorporated.

File: app/models/maestrano/connector/rails/external.rb (hunk 2 above)
Comment (MehreenT) [reply app/models/maestrano/connector/rails/external.rb]: Thankyou. Feedback incorporated.

**Review by MarcoBgn — APPROVED** [review]

**Pull request merged**